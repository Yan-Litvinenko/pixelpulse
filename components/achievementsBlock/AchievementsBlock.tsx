'use client';

import React from 'react';
import AchievementsAchieve from '../achievementsAchieve/AchievementsAchieve';
import styles from '@/styles/components/achievementsBlock/AchievementsBlock.module.scss';
import { nanoid } from '@reduxjs/toolkit';
import type { Rarity, ExecutionStatus, AchievementsBlockProps } from '@/interface/achievements/achievements.interface';

export default function AchievementsBlock(props: AchievementsBlockProps): React.JSX.Element {
    const { achievements, prefixForClassName } = props;

    if (!achievements.length) {
        return <></>;
    }

    return (
        <div className={styles[prefixForClassName]}>
            <span className={styles[`${prefixForClassName}__title`]}>
                {prefixForClassName === 'achieved' ? 'achieved:' : 'to be achieved:'}
            </span>
            <div className={styles[`${prefixForClassName}__achievements`]}>
                {achievements.map((achieve) => (
                    <AchievementsAchieve
                        date={achieve.date}
                        description={achieve.description}
                        key={nanoid()}
                        rarity={achieve.rarity as Rarity}
                        executionStatus={achieve.status as ExecutionStatus}
                        title={achieve.title}
                    />
                ))}
            </div>
        </div>
    );
}
