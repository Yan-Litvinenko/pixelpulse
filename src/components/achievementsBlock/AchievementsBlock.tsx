import React from 'react';
import styles from './AchievementsBlock.module.scss';
import { AchievementsAchieve } from '../achievementsAchieve/AchievementsAchieve';
import { nanoid } from '@reduxjs/toolkit';
import type {
    IAchievementsBlock,
    ExecutionStatus,
    Rarity,
} from '../../interfaces/interface.achievements';

const AchievementsBlock = (props: IAchievementsBlock): React.JSX.Element => {
    const { achievements, prefixForClassName } = props;

    return (
        <>
            {achievements.length ? (
                <div
                    className={styles[prefixForClassName]}
                    data-testId={'achievements-block'}
                >
                    <span className={styles[`${prefixForClassName}__title`]}>
                        {prefixForClassName === 'achieved'
                            ? 'achieved:'
                            : 'to be achieved:'}
                    </span>
                    <div
                        className={
                            styles[`${prefixForClassName}__achievements`]
                        }
                    >
                        {achievements.map((achieve) => (
                            <AchievementsAchieve
                                date={achieve.date}
                                description={achieve.description}
                                key={nanoid()}
                                rarity={achieve.rarity as Rarity}
                                executionStatus={
                                    achieve.status as ExecutionStatus
                                }
                                title={achieve.title}
                            />
                        ))}
                    </div>
                </div>
            ) : (
                <></>
            )}
        </>
    );
};

export { AchievementsBlock };
