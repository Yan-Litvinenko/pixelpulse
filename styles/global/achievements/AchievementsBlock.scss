@import '@/styles/mixin.scss';

.ongoing_block {
    &__title {
        display: block;
        margin-bottom: 1.8rem;
    }

    &__achievements {
        display: flex;
        flex-direction: column;
        gap: 2rem 0rem;
    }
}

.achieved_block {
    &__title {
        display: block;
        margin-bottom: 1.8rem;
    }

    &__achievements {
        display: flex;
        flex-direction: column;
        gap: 2rem 0rem;
    }
}
