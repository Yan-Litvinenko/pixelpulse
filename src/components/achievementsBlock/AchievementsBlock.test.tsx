import React from 'react';
import { AchievementsBlock } from './AchievementsBlock';
import { render, screen } from '@testing-library/react';
import type { IAchieve } from '../../interfaces/interface.achievements';

const achieve: IAchieve[] = [
    {
        date: '2024.01.23',
        description: 'Learn webpack',
        rarity: 'epic',
        status: 'achieved',
        title: 'Professional tools',
    },
];

describe('AchievementsBlock component', (): void => {
    test('Render with achieved length greater than 0', (): void => {
        const component = render(<AchievementsBlock achievements={achieve} prefixForClassName="achieved" />);
        const block = screen.getByTestId('achievements-block');

        expect(block).toBeInTheDocument();
        expect(component).toMatchSnapshot();
    });

    test('Render with array length equal to 0', (): void => {
        render(<AchievementsBlock achievements={[]} prefixForClassName="achieved" />);
        const block = screen.queryByTestId('achievements-block');

        expect(block).toBeNull();
    });
});
