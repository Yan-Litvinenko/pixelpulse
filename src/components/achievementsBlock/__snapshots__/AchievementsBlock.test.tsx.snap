// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AchievementsBlock component Render with achieved length greater than 0 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        data-testid="achievements-block"
      >
        <span>
          achieved:
        </span>
        <div>
          <article
            class="undefined undefined "
          >
            <figure>
              <img
                alt="Rarity icon representing epic"
                src="test-file-stub"
              />
              <figcaption>
                epic
              </figcaption>
            </figure>
            <div
              class="undefined"
            >
              <div>
                <h4>
                  Professional tools
                   
                </h4>
                <p>
                  Learn webpack
                </p>
                <time
                  datetime="2024.01.23"
                >
                  achieved:
                  2024.01.23
                </time>
              </div>
            </div>
          </article>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      data-testid="achievements-block"
    >
      <span>
        achieved:
      </span>
      <div>
        <article
          class="undefined undefined "
        >
          <figure>
            <img
              alt="Rarity icon representing epic"
              src="test-file-stub"
            />
            <figcaption>
              epic
            </figcaption>
          </figure>
          <div
            class="undefined"
          >
            <div>
              <h4>
                Professional tools
                 
              </h4>
              <p>
                Learn webpack
              </p>
              <time
                datetime="2024.01.23"
              >
                achieved:
                2024.01.23
              </time>
            </div>
          </div>
        </article>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
